modules/javafx.controls/src/main/java/javafx/scene/control/skin/TabPaneSkin.java

```diff
@@ -647,7 +645,9 @@ private void initializeTabListener() {
                     }
                 }
 
-                addTabs(tabsToAdd, insertPos == -1 ? tabContentRegions.size() : insertPos);
+                if (!tabsToAdd.isEmpty()) {
+                    addTabs(tabsToAdd, getSkinnable().getTabs().indexOf(tabsToAdd.get(0)));
```

Comment:
@kleopatra is right in that there is no guarantee that the set of added sublists from multiple Changes are contiguous. Since this assumption is preexisting, I agree that this is just something to keep in mind.